docs/content/docs/dev/table/sql/queries/hints.md

```diff
@@ -84,4 +84,205 @@ insert into kafka_table1 /*+ OPTIONS('sink.partitioner'='round-robin') */ select
 
 ```
 
+## Query Hints
+
+`Query Hints` specify that the indicated hints are used in the scope of a query. They affect all operators in the statement.
+Now, Flink `Query Hints` support `Join Hints`.
+
+### Syntax
+The `Query Hints` syntax in Flink:
```

Comment:
we can mention that the query hints syntax follows Apache Calcite

Reply:
> we can mention that the query hints syntax follows Apache Calcite

Done!